lib/matplotlib/backends/backend_pgf.py

```diff
@@ -990,4 +1009,217 @@ def _cleanup_all():
+        self._file.write(rb'\end{document}\n')
+        self._file.close()
+
+        if self._n_figures > 0:
+            try:
+                self._run_latex()
+            finally:
+                try:
+                    shutil.rmtree(self._tmpdir)
+                except:
+                    TmpDirCleaner.add(self._tmpdir)
+        elif self.keep_empty:
+            open(self._outputfile, 'wb').close()
+
+    def _run_latex(self):
+        texcommand = get_texcommand()
+        cmdargs = [
+            str(texcommand),
+            "-interaction=nonstopmode",
+            "-halt-on-error",
+            os.path.basename(self._fname_tex),
+        ]
+        try:
+            subprocess.check_output(
+                cmdargs, stderr=subprocess.STDOUT, cwd=self._tmpdir
+            )
+        except subprocess.CalledProcessError as e:
+            raise RuntimeError(
+                "%s was not able to process your file.\n\nFull log:\n%s"
+                % (texcommand, e.output.decode('utf-8')))
+
+        # copy file contents to target
+        with open(self._fname_pdf, "rb") as fh_src:
+            with open(self._outputfile, "wb") as fh:
+                shutil.copyfileobj(fh_src, fh)
+
+    def savefig(self, figure=None, **kwargs):
+        """
+        Saves a :class:`~matplotlib.figure.Figure` to this file as a new page.
+
+        Any other keyword arguments are passed to
+        :meth:`~matplotlib.figure.Figure.savefig`.
+
+        Parameters
+        ----------
+
+        figure : :class:`~matplotlib.figure.Figure` or int, optional
+            Specifies what figure is saved to file. If not specified, the
+            active figure is saved. If a :class:`~matplotlib.figure.Figure`
+            instance is provided, this figure is saved. If an int is specified,
+            the figure instance to save is looked up by number.
+        """
+        if not isinstance(figure, Figure):
+            if figure is None:
+                manager = Gcf.get_active()
+            else:
+                manager = Gcf.get_fig_manager(figure)
+            if manager is None:
+                raise ValueError("No figure {}".format(figure))
+            figure = manager.canvas.figure
+
+        try:
+            orig_canvas = figure.canvas
+            figure.canvas = FigureCanvasPgf(figure)
+
+            if self._n_figures == 0:
+                self._write_header(*figure.get_size_inches())
+            else:
+                if get_texcommand() == 'lualatex':
+                    if _get_lualatex_version() > (0, 85, 0):
+                        np = r'\newpage\pagewidth={}in\pageheight={}in%'
+                    else:
+                        np = r'\newpage\pdfpagewidth={}in\pdfpageheight={}in%'
+                else:
+                    np = r'\newpage\pdfpagewidth={}in\pdfpageheight={}in%'
+                self._file.write(np.format(
+                        *figure.get_size_inches()
+                    ).encode('utf-8') + b'\n'
```

Comment:
just include the newline in the format-string above?
here it may make sense to just use a bytes literal and %-interpolation